rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java

```diff
@@ -219,34 +259,44 @@ public boolean removeCommitted(Uid lraUid) {
 
     public void getFailedLRAs(Map<URI, LongRunningAction> lras) {
         InputObjectState aa_uids = new InputObjectState();
-        Consumer<Uid> failedLRACreator = uid -> {
-            LongRunningAction lra = new LongRunningAction(lraService, new Uid(uid));
-            lra.activate();
+        String failedTransactionType = LongRunningAction.getFailedLraType();
 
-            LRAStatus status = lra.getLRAStatus();
-            if (LRAStatus.FailedToCancel.equals(status) || LRAStatus.FailedToClose.equals(status)) {
-                lras.put(lra.getId(), lra);
-            }
-        };
+        Consumer<Uid> failedLRACreator = getFailedLRACreator(lras, true);
```

Comment:
We just need a list of the records, ie we don't need to activate the record. The following should be sufficient:

``` java
_recoveryStore.allObjUids("/StateManager/BasicAction/LongRunningAction/Failed", <input buffer);
```

Reply:
Now above comments will change the implementation here.

Reply:
Okay I will review that change when you next push an update to this PR, thanks. If we can keep the changes as minimal as possible then we can reduce opportunities for bugs to creep in (the ratio of bugs to new lines of code is still quite high in our industry).

Reply:
I have pushed new changes. Here almost the code is same a before, just I did some code refactoring, and checking the failed record in both the location.